optax/_src/base_test.py

```diff
@@ -59,5 +63,44 @@ def test_set_to_zero_is_stateless(self):
         self.variant(base.set_to_zero().init)(params=None), base.EmptyState())
 
 
+class StatelessTest(chex.TestCase):
+  """Tests for the stateless transformation."""
+
+  @chex.all_variants
+  @parameterized.parameters(False, True)
+  def test_stateless(self, on_leaves):
+    params = {'a': jnp.zeros((1, 2)), 'b': jnp.ones((1,))}
+    updates = {'a': jnp.ones((1, 2)), 'b': jnp.full((1,), 2.0)}
+
+    if on_leaves:
+      def weight_decay(g, p):
+        return g + 0.1 * p
+    else:
+      def weight_decay(g, p):
+        return jax.tree_map(lambda g_, p_: g_ + 0.1 * p_, g, p)
+
+    opt = base.stateless(weight_decay, on_leaves=on_leaves)
+    state = opt.init(params)
```

Comment:
I think we could guarantee to the user that `stateless` accepts and returns an `EmptyState()`. This could be done by using `EmptyState()` directly in the update function rather than calling `init` and introducing a very simple separate test for `init` (just checking it returns EmptyState).

Reply:
Agreed and done!